test/models/TradingDetailsSpec.scala

```diff
@@ -25,14 +25,8 @@ class TradingDetailsSpec extends PlaySpec {
 
     "have no errors" when {
 
-      "both fields are correct - lowercase z in isa ref" in {
```

Comment:
I'd have preferred us to keep this test but change the check to be that res.errors should not be Nil. That way we have confidence that lowercase z isn't being treated as valid. (This is me assuming there isn't already a test doing this).

Reply:
added test case for invalid data i.e lowercase z

Reply:
I think a test case needs to be added in this file too and not just at the controller level!